Add eslint.js for WebStorm compat

zurawiki opened the pull request:
By adding a symlink, we can take advantage of the eslint_d inside of the WebStorm IDE

This does not add eslint to the npm bin, but just make sure `eslint.js` exists for tools and plugins that look for it.

Add eslint_d to WebStorm's ESLint config
![WebStorm ESLint Config](https://cloud.githubusercontent.com/assets/1464035/11602524/c68ff9ee-9a8f-11e5-87b0-75d5ba865f41.png)

Without this patch, we get the following error.
![WebStorm Error Message](https://cloud.githubusercontent.com/assets/1464035/11602523/c68ef4ae-9a8f-11e5-855b-8a5300100995.png)

With the patch, the linter is faster than ever!

mantoni commented:
Thanks for the pull request. This means that WebStorm is doing the wrong thing. They shouldn't make assumptions about the internal structure of the ESLint package. But the workaround is easy enough, so yeah, we can add this.

Two minor things though: If you want to add a screenshot, please make it part of the PR and link to it with a relative URL to the image instead of some hash. Alternatively, a text description would also do. Secondly, please replace the compatibility note with something like "You need at least eslint_d 2.4.0 for WebStorm support." instead of linking to your fork - that branch in your fork will most likely not receive future patches ;-)

Thanks!

mantoni commented on bin/eslint.js:
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+var client = require('../lib/client');
+var useStdIn = (process.argv.indexOf('--stdin') > -1);
```
The stdin option is already supported. This file could simply `require('./eslint_d.js');`.

zurawiki commented:
I updated the PR and simplified the readme. I am not sure what version you will publish so if you want to add a reference to the version number, I will give you the liberty to do so.

mantoni commented:
Thanks! Released in `2.4.0`. :ship:

SimenB commented:
This is broken again in newer IntelliJs.

`ESLint error:: Initialization error (ESLint). Cannot find module '/Users/simen/Development/add-asset-html-webpack-plugin/node_modules/eslint_d/lib/cli.js'`

mantoni commented:
@SimenB D'oh! Do you know how to solve it?

SimenB commented:
I don't think it's possible, but the newer intellij uses the api more directly out of the box, so not as needed as before, maybe?

https://blog.jetbrains.com/webstorm/2017/02/webstorm-2017-1-eap-171-3224/

I still use eslint_d for auto-fix on save, though :smile:

mantoni commented:
Atom always did it this way and it absolutely makes sense for IDEs. I think we don't habe to open a new issue then. Would you care to put a note into the Readme stating the version range where eslint_d works?
